### PR TITLE
Hw tx validation

### PR DESCRIPTION
**Pull request by Riddlerrr**
tasks:
- https://www.pivotaltracker.com/story/show/177083008
- https://www.pivotaltracker.com/story/show/177272254

### CODE REVIEW

**Comment by Riddlerrr** [comment]
@aquabu @Alexey1100 Could you review the PR and merge it if all good?

File: hotwallet/lib/hotwalletUtils.js [hunk 1]
```diff
@@ -398,17 +430,19 @@ exports.waitForNewTransaction = async function waitForNewTransaction(envs, hwRed
...
-  const hasTokens = await hwAlgorand.positiveTokenBalance(hwAddress, envs.optInApp)
+  const hasTokens = await hwAlgorand.positiveTokenBalance(hwAddress)
   if (!hasTokens) {
     console.log(`The Hot Wallet does not have NOTE tokens. Please top up the ${hwAddress}`)
     return false
   }
 
+  // hwAlgorand
   console.log("Checking for a new transaction to send...")
   const hwApi = new ComakeryApi(envs)
```
Comment (Alexey1100) [inline hotwallet/lib/hotwalletUtils.js:441]: Interesting how it's not throwing an error, when redefining this constant later on line 449

**Review by Alexey1100 — APPROVED** [review]
👍

**Review by aquabu — APPROVED** [review]

File: hotwallet/lib/hotwalletUtils.js (hunk 1 above)
Comment (Riddlerrr) [reply hotwallet/lib/hotwalletUtils.js:441]: 👍 Good find. Will delete the second initialization

**Pull request merged**